### PR TITLE
[WIP] feat: Provide installation instructions on missing language preprocessors

### PR DESCRIPTION
**Pull request by mikemenaker**

### CODE REVIEW

File: lib/compileTemplate.ts [hunk 1]
```diff
@@ -40,6 +40,16 @@ export function compileTemplate (
...
     }))
+  } else if (preprocessLang) {
+    return {
+      code: (
+        `var render = function () {}\n` +
+        `var staticRenderFns = []\n`
+      ),
+      source: options.source,
+      tips: [`Please install ${preprocessLang}`],
+      errors: [`Please install ${preprocessLang}`]
```
Comment (znck) [inline lib/compileTemplate.ts]: The error message should tell what is not working.
And then the tip can suggest how to fix it.

Comment (mikemenaker) [reply lib/compileTemplate.ts]: Updated, wanted to make sure I had my logic correct first

File: test/compileTemplate.spec.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    filename: 'example.vue',
+    source: template.content,
+    preprocessLang: template.lang,
+    compiler: compiler
+  })
+
+  expect(result.errors.length).toBe(0)
+})
+
+test('preprocess pug', () => {
```
Comment (znck) [inline test/compileTemplate.spec.ts]: The test name can be set to `'warn missing preprocessor'`.

Comment (mikemenaker) [reply test/compileTemplate.spec.ts]: Updated test name

**Pull request merged**